.github/workflows/build.yml

```
@@ -23,35 +23,74 @@ on:
       - '.all-contributorsrc'
 
 jobs:
-  build:
-
+  # ----------------------------------------------------------------------------
+  # 1) JVM build + tests under Java 17
+  # ----------------------------------------------------------------------------
+  build-jvm:
+    name: JVM Build & Tests (Java 17)
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-java@v4
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v4
         with:
           java-version: '17'
           distribution: 'temurin'
-      - uses: graalvm/setup-graalvm@v1
-        with:
-          java-version: '17'
-          distribution: 'graalvm'
-          components: 'native-image'
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          cache: 'maven'
 
-      - name: Get Date
+      - name: Get Date for cache key
         id: get-date
-        run: |
-          echo "::set-output name=date::$(/bin/date -u "+%Y-%m")"
+        run: echo "::set-output name=date::$(date -u '+%Y-%m')"
```

set-output is already [deprecated](https://github.blog/changelog/2022-10-10-github-actions-deprecating-save-state-and-set-output-commands/)
```suggestion
        run: echo "date=$(date -u '+%Y-%m')" >> "$GITHUB_OUTPUT"
```